vertx/src/main/java/com/groupon/MainVerticle.java

```diff
@@ -1,72 +1,56 @@
     private Single<Integer> sumOverNetwork(List<Integer> values) {
         // if (values.size() == 0) return Single.error(new Exception("Cannot sum 0 numbers"));
         // if (values.size() == 1) return Single.just(values.get(0));
-        return Single.fromEmitter(emitter -> {
-            HttpClientRequest request = httpClient.getAbs(Utils.urlForCalc(values));
-            request.putHeader("User-Agent", "Vertx");
-            request.putHeader("Accept-Encoding", "gzip");
-            request.setTimeout(1_000_000);
-
-            request.toObservable()
-                    .flatMap(HttpClientResponse::toObservable)
-                    .map(buffer -> buffer.toString("UTF-8"))
-                    .reduce((s, s2) -> s + s2)
-                    .subscribe(body -> {
-                        int sum = Integer.valueOf(body);
-                        // logger.debug("Received from network: " + sum);
-                        emitter.onSuccess(sum);
-                    }, emitter::onError);
-            request.end();
-        });
+        return Single.fromEmitter(emitter -> sumVerticle.callSumNetwork(values)
+                .subscribe(body -> {
+                    int sum = Integer.valueOf(body);
+                    // logger.debug("Received from network: " + sum);
+                    emitter.onSuccess(sum);
+                }, emitter::onError));
     }
 
     @Override
     public void start(Future<Void> fut) {
-        httpClient = vertx.createHttpClient(
-                new HttpClientOptions()
-                        .setKeepAlive(false)
-                        .setMaxPoolSize(64) // 5 default // this is per host
-                        .setLogActivity(false)
-                        .setMaxWaitQueueSize(-1) // -1 default
-                        //.setPipelining(true)
-                        //.setPipeliningLimit(100)
-                        .setProtocolVersion(HttpVersion.HTTP_1_0) // 1.1 default
-                        .setConnectTimeout(1_000_000)
-                        .setIdleTimeout(1_000) // For some reasons now connections are established
-                        // with 0.
-                        .setTcpNoDelay(true) // Same as in jetty.
-                        .setTryUseCompression(false));
+        vertx.eventBus().getDelegate().registerDefaultCodec(Collection.class, new CollectionMessageCodec<>());
+        sumVerticle = new SumVerticle(vertx);
+        RxHelper.deployVerticle(vertx, sumVerticle);
         Method[] methods = new Method[]{
                 Method.fromMemory(),
                 new Method(Method.METHOD_NAME_NETWORK, this::sumOverNetwork),
+                new Method(Method.METHOD_EVENTBUS_NETWORK, integers -> {
+                    Single<Message<Integer>> messageSingle = vertx.eventBus().rxSend(Method.METHOD_EVENTBUS_NETWORK, integers, new DeliveryOptions().setCodecName(CollectionMessageCodec.NAME));
+                    return messageSingle.flatMap(event -> Single.just(event.body()));
```

Comment:
You should replace with .map() and skip the Single wrapping

Reply:
oh you're right, removed the single wrap